### PR TITLE
Catch psql errors

### PR DESCRIPTION
**Pull request by JaxTurboNerd**
On the server side, set response status = 598 when a unique constraint error exists.  Will check for this status on the fetch call and react with a message on the pop up modal.

### CODE REVIEW

**Comment by JaxTurboNerd** [comment]
Don't merge

File: client/src/components/addName.js [hunk 1]
```diff
@@ -25,12 +28,20 @@ function AddName() {
...
+  const capitalize = (input) => {
+    const value = input.trim();
+    return value.charAt(0).toUpperCase() + value.slice(1);
+  };
+
   const onSubmit = (formData) => {
     // POST API call
-    fetch(`${process.env.REACT_APP_HOST}/names`, {
+    fetch(`http://localhost:3001/names`, {
+      // fetch(`${process.env.REACT_APP_HOST}/names`, {
```
Comment (murjax) [inline client/src/components/addName.js:44]: Switch this back before merging.

File: client/src/components/addName.js [hunk 2]
```diff
@@ -39,15 +50,25 @@ function AddName() {
       body: JSON.stringify(formData),
     })
       .then((response) => {
+        //console.log(response);
```
Comment (murjax) [inline client/src/components/addName.js:53]: Drop this console log comment.

File: client/src/components/main.js [hunk 3]
```diff
@@ -6,7 +6,8 @@ function Main() {
   const [petName, setPetName] = useState();
 
   const getName = useCallback(() => {
-    fetch(`${process.env.REACT_APP_HOST}/names?is_male=${isMale}`)
+    fetch(`http://localhost:3001/names?is_male=${isMale}`)
+      // fetch(`${process.env.REACT_APP_HOST}/names?is_male=${isMale}`)
```
Comment (murjax) [inline client/src/components/main.js:10]: Switch this back too.

File: server/queries.js [hunk 4]
```diff
@@ -33,12 +33,21 @@ const getNameById = async (request, response) => {
...
+    const result = await knex('pet_names').insert({ name, is_male }, [
+      'id',
+      'name',
+      'is_male',
+    ]);
+    response.status(201).json(result[0]);
+  } catch (error) {
+    console.log(error);
+    if (error.code == '23505') {
+      response.status(598).json({ msg: `${name} already exists` });
```
Comment (murjax) [inline server/queries.js:46]: Why 598? This appears to be a network timeout code. https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#598

I usually see 422 used as the response status for validation issues. Any chance of using that or are there any issues with that?

**Review by murjax — COMMENTED** [review]
Few comments, but glad to see some better handling of the constraint errors!

File: client/src/components/addName.js (hunk 1 above)
Comment (JaxTurboNerd) [reply client/src/components/addName.js:44]: I will switch back.  I was using my local docker container to test these changes.

File: client/src/components/addName.js (hunk 2 above)
Comment (JaxTurboNerd) [reply client/src/components/addName.js:53]: done!

File: server/queries.js (hunk 4 above)
Comment (JaxTurboNerd) [reply server/queries.js:46]: changed to 422.  I just picked a random response status code for testing.

**Review by murjax — APPROVED** [review]
Good to go!

**Pull request merged**